### PR TITLE
Feature/housing user likes

### PR DESCRIPTION
**Pull request by nfilzi**

### CODE REVIEW

File: apps/web/config/routes.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@
     member do
       delete :dismiss
     end
+    resource :likes, only: [:create, :destroy]
```
Comment (cveneziani) [inline apps/web/config/routes.rb]: missing empty line in between

File: db/migrations/20180408123120_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  change do
+    create_table :likes do
+      primary_key :id
+      foreign_key :housing_id, :housings, null: false, on_delete: :cascade
+      foreign_key :user_id,    :users,    null: false, on_delete: :cascade
+
+      column :created_at, DateTime, null: false
+      column :updated_at, DateTime, null: false
+
+
```
Comment (cveneziani) [inline db/migrations/20180408123120_create_likes.rb]: useless extra empty line

File: lib/housing_list/entities/housing_with_likes_count.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @likes_count      = attributes[:likes_count] || 0
+    attributes[:user] = build_user(attributes)
+
+    super(attributes)
+  end
+
+  private
+
+  def build_user(attributes)
+    user_attribute_names = attributes.keys.select { |key| key.match /^users_.*$/ }
```
Comment (cveneziani) [inline lib/housing_list/entities/housing_with_likes_count.rb]: maybe we should comment a bit what happens here, for the future ourselves

File: lib/housing_list/interactors/likes/destroy.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+module Likes
+  class Destroy
+    include Hanami::Interactor
+
+    private
+    attr_reader :like
+
+    public
+    def initialize(user, housing)
+      @like = LikeRepository.new.find_for_user_and_housing(user.id, housing.id)
```
Comment (cveneziani) [inline lib/housing_list/interactors/likes/destroy.rb]: I'm not found of an initializer that already launches a search.
I prefer to have in into the `call` method.

What do you think?

File: lib/housing_list/repositories/housing_repository.rb [hunk 5]
```diff
@@ -1,5 +1,6 @@
 class HousingRepository < Hanami::Repository
   associations do
+    has_many :likes
```
Comment (cveneziani) [inline lib/housing_list/repositories/housing_repository.rb]: `has_many` should be placed after `belongs_to` 
ref: https://github.com/bbatsov/rails-style-guide#macro-style-methods (even if it's a rails styleguide :grimacing:)

File: lib/housing_list/repositories/housing_repository.rb [hunk 6]
```diff
@@ -13,17 +14,36 @@ def find_with_user(id)
   end
 
   def active_for_trip_sorted_by_most_recent(trip_id)
```
Comment (cveneziani) [inline lib/housing_list/repositories/housing_repository.rb]: shall we rename this method? (Idem lines 21 and 28)

File: lib/housing_list/interactors/likes/destroy.rb (hunk 4 above)
Comment (nfilzi) [reply lib/housing_list/interactors/likes/destroy.rb]: In this case, it feels to me that the `like` retrieval process is a kind of initialisation process for our interactor. It also feel like an interactor used to delete a like should not be bothered with `housing` and `user` as state. What do you think?

**Pull request merged**